test/Microsoft.AspNetCore.SignalR.Client.Tests/RpcConnectionFacts.cs

```diff
@@ -0,0 +1,127 @@
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Client.Tests
+{
+    public class RpcConnectionFacts : IDisposable
+    {
+        TestServer testServer;
+        public RpcConnectionFacts()
+        {
+            var webHostBuilder = new WebHostBuilder().
+                ConfigureServices(services =>
+                {
+                    services.AddSignalR();
+                })
+                .Configure(app =>
+                {
+                    app.UseSignalR(routes =>
+                    {
+                        routes.MapHub<TestHub>("/hubs");
+                    });
+                });
+            testServer = new TestServer(webHostBuilder);
+        }
+
+        [Fact]
+        public async void CheckFixedMessage()
+        {
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = testServer.CreateClient())
+            using (var pipelineFactory = new PipelineFactory())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await HubConnection.ConnectAsync(new Uri("http://test/hubs"), new JsonNetInvocationAdapter(), transport, httpClient, pipelineFactory, loggerFactory))
+                {
+
+                    var result = await connection.Invoke<string>("Microsoft.AspNetCore.SignalR.Client.Tests.RpcConnectionFacts+TestHub.HelloWorld");
+
+                    Assert.Equal("Hello World!", result);
+                }
+            }
+        }
+
+        [Fact]
+        public async void CheckEchoMessage()
+        {
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = testServer.CreateClient())
+            using (var pipelineFactory = new PipelineFactory())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await HubConnection.ConnectAsync(new Uri("http://test/hubs"), new JsonNetInvocationAdapter(), transport, httpClient, pipelineFactory, loggerFactory))
+                {
+                    var result =  await connection.Invoke<string>("Microsoft.AspNetCore.SignalR.Client.Tests.RpcConnectionFacts+TestHub.Echo", "SignalR");
+
+                    Assert.Equal("SignalR", result);
+                }
+            }
+        }
+
+        [Fact]
+        public async void CheckCallEcho()
+        {
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = testServer.CreateClient())
+            using (var pipelineFactory = new PipelineFactory())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await HubConnection.ConnectAsync(new Uri("http://test/hubs"), new JsonNetInvocationAdapter(), transport, httpClient, pipelineFactory, loggerFactory))
+                {
+                    var manualresetEvent = new ManualResetEvent(false);
```

Comment:
We did it all over the place in the old client. One of the reasons was that old xUnit did not handle async tests. Now that it does I find [the pattern](https://github.com/aspnet/SignalR-Server/blob/beb81a575ed51ecb375c046fd6b2e2abe30e94c1/test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTestClient.cs#L69-L74) @anurse used in the compat tests much nicer.